src/curve.c

```diff
@@ -152,6 +152,218 @@ int ec_public_key_serialize_protobuf(ProtobufCBinaryData *buffer, const ec_publi
     return 0;
 }
 
+int alice_s_buf_serialize_protobuf(ProtobufCBinaryData *buffer, const signal_buffer *alice_s_buf)
+{
+    size_t len = 0;
+    uint8_t *data = 0;
+
+    assert(buffer);
+    assert(alice_s_buf);
+
+    len = sizeof(uint8_t) * (DJB_KEY_LEN);
+    data = malloc(len);
+    if(!data) {
+        return SG_ERR_NOMEM;
+    }
+
+    //data[0] = DJB_TYPE;
+    memcpy(data, alice_s_buf->data, DJB_KEY_LEN);
+
+    buffer->data = data;
+    buffer->len = len;
+
+    /*int i;
+    printf("stored serialized s buf data\n");
+    for(i = 0; i < DJB_KEY_LEN; i++){
+        printf("%hhu\n", data[i]);
+    }*/
+
+    return 0;
+}
+
+int alice_s_buf_deserialize_protobuf(signal_buffer **s_buf, const uint8_t *s_buf_data, size_t s_buf_len, signal_context *global_context)
```

Comment:
It seems like in these deserialize functions, we do not really need to pass `s_buf_len` (since we are hard coding that value later on anyways-- see line 200 for example) or `global_context` (not even used)

Reply:
removed `global_context`.
length parameters are necessary here as the deserialize functions are unwrapping `Protobuf` datatypes and storing them in `signal_buffer` structs.